flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -169,6 +181,83 @@ private static void createMaterializedInContinuousMode(
         }
     }
 
+    private static ResultFetcher callAlterMaterializedTableRefreshOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            AlterMaterializedTableRefreshOperation alterMaterializedTableRefreshOperation) {
+        ObjectIdentifier materializedTableIdentifier =
+                alterMaterializedTableRefreshOperation.getTableIdentifier();
+        ResolvedCatalogBaseTable<?> table = operationExecutor.getTable(materializedTableIdentifier);
+        if (!(table instanceof ResolvedCatalogMaterializedTable)) {
+            throw new TableException(
```

Comment:
I think we should unify the exception type to `SqlExecutionException` in SqlGateway.

Reply:
Got it!